core/src/data/dataSource.cpp

```diff
@@ -120,23 +128,15 @@ void DataSource::constructURL(const TileID& _tileCoord, std::string& _url) const
     }
 }
 
-bool DataSource::getTileData(std::shared_ptr<TileTask>& _task) {
-    if (m_cache->get(_task)) {
-        _task->loaded = true;
-        return true;
-    }
-    return false;
-}
-
 void DataSource::onTileLoaded(std::vector<char>&& _rawData, std::shared_ptr<TileTask>& _task, TileTaskCb _cb) {
-    TileID tileID = _task->tile->getID();
+    TileID tileID = _task->tileId();
 
     if (!_rawData.empty()) {
-        _task->loaded = true;
 
         auto rawDataRef = std::make_shared<std::vector<char>>();
         std::swap(*rawDataRef, _rawData);
-        _task->rawTileData = rawDataRef;
+
+        dynamic_cast<DownloadTileTask*>(_task.get())->rawTileData = rawDataRef;
```

Comment:
In this case it's also safe to use a `static_cast` since it happens to be only for downloadable tiles. We should think of reducing usage of `dynamic_cast` for no RTTI in general anyway.